packages/browser/src/core/analytics/index.ts

```diff
@@ -434,7 +411,9 @@ export class Analytics
     ) as LegacyDestination[]
 
     legacyDestinations.forEach((destination) => {
-      destination.addMiddleware(...middlewares)
+      if ('addMiddleware' in destination) {
```

Comment:
This check technically might not be necessary, but it was revealed when better types were added.